src/net_processing.cpp

```diff
@@ -2305,8 +2305,11 @@ bool PeerManagerImpl::AlreadyHave(const CInv& inv)
         return m_llmq_ctx->clhandler->AlreadyHave(inv);
     case MSG_ISDLOCK:
         return m_llmq_ctx->isman->AlreadyHave(inv);
+    case MSG_DSQ:
+        return m_cj_ctx->server->HasQueue(inv.hash);
```

Comment:
should check queueman (client-side) too I guess
```suggestion
        return m_cj_ctx->server->HasQueue(inv.hash) || m_cj_ctx->queueman->HasQueue(inv.hash);
```

Reply:
I thought about that, but queue man only exists if the wallet is compiled in, and the both should see it if it exists no?

Reply:
client-side cj works on non-masternodes only (with wallets compiled in), server-side cj works on masternodes only.

Reply:
check `ProcessMessage` in both modules